### PR TITLE
[SPARK-28745][SQL][TEST] Add benchmarks for `extract()`

### PR DESCRIPTION
**Pull request by MaxGekk**
## What changes were proposed in this pull request?

Added new benchmark `ExtractBenchmark` for the `EXTRACT(field FROM source)` function. It was executed on all currently supported values of the `field` argument:  `MILLENNIUM`, `CENTURY`, `DECADE`, `YEAR`, `ISOYEAR`, `QUARTER`, `MONTH`, `WEEK`, `DAY`, `DAYOFWEEK`, `HOUR`, `MINUTE`, `SECOND`, `MILLISECONDS`, `MICROSECONDS`, `EPOCH`. The `cast(id as timestamp)` was taken as the `source` argument.

## How was this patch tested?

By running the benchmark via:
```
$ SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.execution.benchmark.ExtractBenchmark"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@dongjoon-hyun Please, take a look at the benchmarks.

**Comment by dongjoon-hyun** [comment]
Thank you for pinging me, @MaxGekk . Sure!

File: sql/core/benchmarks/ExtractBenchmark-results.txt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+SECOND of timestamp:                      Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------------------------------
+SECOND of timestamp wholestage off                  408            424          22         24.5          40.8       1.0X
+SECOND of timestamp wholestage on                   401            405           3         24.9          40.1       1.0X
+
+Java HotSpot(TM) 64-Bit Server VM 1.8.0_202-b08 on Mac OS X 10.14.6
+Intel(R) Core(TM) i7-4850HQ CPU @ 2.30GHz
+MILLISECONDS of timestamp:                Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------------------------------
+MILLISECONDS of timestamp wholestage off          29329          30007         960          0.3        2932.9       1.0X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/ExtractBenchmark-results.txt]: `Steve = 960` ?

**Comment by dongjoon-hyun** [comment]
Hi, @MaxGekk .
I regenerated the result on EC2 and made a PR to you. Could you review and merge that?
- https://github.com/MaxGekk/spark/pull/19

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @MaxGekk .

This is only adding additional benchmark code. I tested this in EC2 machine in order to generate the result.

Merged to master.

**Pull request closed without merging**